src/SFA.DAS.Activities/IntegrityChecker/Utils/CosmosClient.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Documents.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using SFA.DAS.Activities.Configuration;
+using SFA.DAS.Activities.IntegrityChecker.Interfaces;
+
+namespace SFA.DAS.Activities.IntegrityChecker.Utils
+{
+    class CosmosClient : ICosmosClient
+    {
+        private readonly ICosmosConfiguration _config;
+        private readonly IDocumentCollectionConfigurator _documentCollectionConfigurator;
+        private readonly Lazy<DocumentClient> _client;
+        private readonly ConcurrentDictionary<string, Uri> _collections = new ConcurrentDictionary<string, Uri>();
+        private RequestOptions _requestOptions;
+
+        public CosmosClient(ICosmosConfiguration messageServiceBusConfiguration, IDocumentCollectionConfigurator documentCollectionConfigurator)
+        {
+            _client = new Lazy<DocumentClient>(InitialiseClient);
```

Comment:
It looks like `DocumentClient` won't be disposed when initialized this way. StructureMap supports `Lazy<T>` though and will also call `Dispose()` on any classes implementing `IDisposable` when the container itself is disposed. So something like this should work:

In your container configuration:
```
c.For<DocumentClient>().Use(() => InitializeDocumentClient()).Singleton();
```

In your constructor:
```
public CosmosClient(Lazy<DocumentClient> documentClient)
{
}
```

Alternatively you could implement `IDisposable` on `CosmosClient` and dispose of `DocumentClient` there.

Reply:
Done